perfkitbenchmarker/providers/profitbricks/profitbricks_virtual_machine.py

```diff
@@ -0,0 +1,368 @@
+    else:
+      try:
+          hardware = ast.literal_eval(self.machine_type)
+          logging.info('Using custom hardware configuration from flag.')
+          self.ram = hardware['ram']
+          self.cores = hardware['cores']
+      except ValueError:
+          # ast.literal_eval() raises a ValueError if string can not
+          # be converted to a dict.  This means machine_type is a
+          # preset, i.e. Small, Medium, Large, etc.
+          logging.info('Using preset hardware configuration.')
+          self.ram, self.cores = util.ReturnFlavor(self.machine_type)
+
+  @classmethod
+  def _GetOptionDecoderConstructions(cls):
+    """Gets decoder classes and constructor args for each configurable option.
+
+    Returns:
+      dict. Maps option name string to a (ConfigOptionDecoder class, dict) pair.
+          The pair specifies a decoder class and its __init__() keyword
+          arguments to construct in order to decode the named option.
+    """
+    result = super(ProfitBricksVmSpec, cls)._GetOptionDecoderConstructions()
+    result.update({
+        'machine_type': (MachineTypeDecoder, {}),
+        'profitbricks_location': (option_decoders.StringDecoder,
+                                  {'default': 'us/las'}),
+        'profitbricks_boot_volume_type': (option_decoders.StringDecoder,
+                                          {'default': 'HDD'}),
+        'profitbricks_boot_volume_size': (option_decoders.IntDecoder,
+                                          {'default': 10,
+                                           'min': 10})})
+    return result
+
+
+class ProfitBricksVirtualMachine(virtual_machine.BaseVirtualMachine):
+    """Object representing a ProfitBricks Virtual Machine."""
+
+    CLOUD = providers.PROFITBRICKS
+    DEFAULT_IMAGE = None
+
+    def __init__(self, vm_spec):
+        """Initialize a ProfitBricks virtual machine.
+
+        Args:
+        vm_spec: virtual_machine.BaseVirtualMachineSpec object of the vm.
+        """
+        super(ProfitBricksVirtualMachine, self).__init__(vm_spec)
+
+        # Get user authentication credentials
+        user_config_path = os.path.expanduser(FLAGS.profitbricks_config)
+
+        with open(user_config_path) as f:
+            user_creds = f.read().rstrip('\n')
+            self.user_token = base64.b64encode(user_creds)
+
+        self.server_id = None
+        self.server_status = None
+        self.dc_id = None
+        self.dc_status = None
+        self.lan_id = None
+        self.lan_status = None
+        self.max_local_disks = 1
+        self.local_disk_counter = 0
+        self.ram = vm_spec.ram
+        self.cores = vm_spec.cores
+        self.machine_type = vm_spec.machine_type
+        self.image = self.image or self.DEFAULT_IMAGE
+        self.boot_volume_type = FLAGS.profitbricks_boot_volume_type
+        self.boot_volume_size = FLAGS.profitbricks_boot_volume_size
+        self.location = FLAGS.profitbricks_location
```

Comment:
Same goes for location, but there aren't many use cases right now for running benchmarks with VMs in different locations other than simple networking tests.